src/shared/Core.Tests/ProcessManagerTests.cs

```diff
@@ -0,0 +1,20 @@
+using GitCredentialManager;
+using Xunit;
+
+namespace Core.Tests;
+
+public class ProcessManagerTests
+{
+    [Theory]
+    [InlineData("", 0)]
+    [InlineData("foo", 0)]
+    [InlineData("foo/bar", 1)]
+    [InlineData("foo/bar/baz", 2)]
+    public void TryGetProcessDepth_Returns_Expected_Depth(string input, int expected)
+    {
+        ProcessManager.Sid = input;
+        var actual = ProcessManager.GetProcessDepth();
+
+        Assert.Equal(expected, actual);
+    }
+}
```

Comment:
Just for completeness, do we care if the input string (ie, the SID that we got from the environment) has a leading or trailing slash?

I'm not saying it should happen, but accidents do...

Reply:
Sure - I actually ended up adding a `Trim` in `CreateSid` - felt better to clean up the Sid prior to checking the process depth.